test/tests/bcs/ad_vector_function_neumann_bc/tests

```diff
@@ -4,7 +4,13 @@
         type = 'Exodiff'
         input = 'vector_ad_neumann_bc.i'
         exodiff = 'vector_ad_neumann_bc_out.e'
-        requirement = 'We shall be able to produce the expected result using ADIntegratedBC'
+        requirement = 'The system shall support Neumann type boundary conditions defined by a function for vector variables'
```

Comment:
This requirement wording revision is good, but considering you used `detail` in the Jacobian test, I would suggest doing this instead because using `requirement` in the first test and not in the top level of the `Tests` block might cause an SQA error. 

```
[Tests]
  issues = '#22327'
  requirement = 'The system shall support Neumann type boundary conditions defined by a function'
  [./test]
    ...
    detail = 'for vector variables with automatic differentiation and'
    ...
  [../]
  [./test-jac]
    ...
    detail = 'be able to produce the exact Jacobian.'
  [../]
[]
```

Reply:
Thanks for the help with the test, I was a bit confused.